### ISSUE TITLE
observerBatching presets always bundle CommonJS

### ISSUE THREAD

**Issue by davezuko** [mobxjs/mobx-react-lite#296]
Hey! Love this library, and wanted to let you know about something I discovered today while trimming down our bundles:

When using the side-effect imports to configure observer batching (e.g. `import "mobx-react-lite/batchingForReactDom"`), mobx-react-lite imports the CommonJS module regardless of the preferred module type. This is problematic because mobx-react-lite gets bundled twice unless you happen to be using CommonJS. The offending code:

```js
// batchingForReactDOM.js
const ReactDOM = require("react-dom")
require("./dist").observerBatching(ReactDOM.unstable_batchedUpdates)

// ./dist resolves to ./dist/index.js, which contains:
if (process.env.NODE_ENV === 'production') {
  module.exports = require('./mobxreactlite.cjs.production.min.js')
} else {
  module.exports = require('./mobxreactlite.cjs.development.js')
}
```

You can circumvent this issue by just configuring it yourself, which to me seems preferable anyways since it doesn't rely on side-effects or further complication to the packaging infrastructure.

```ts
import * as ReactDOM from "react-dom"
import {observerBatching} from "mobx-react-lite"
observerBatching(ReactDOM.unstable_batchedUpdates)
```

In my case, this change saved 3kb (~50%) from mobx-react-lite-related imports.
 
## Intended outcome

Importing "mobx-react-lite/batchingForReactDom" in an ESM bundler should import and use `observerBatching` from the same module that "mobx-react-lite" provides in the current environment.

webpack-bundle-analyzer in production:

![image](https://user-images.githubusercontent.com/6439050/88342257-b0eb5f00-cd04-11ea-8c8c-4ab318910e29.png)

## Actual outcome

Importing "mobx-react-lite/batchingForReactDom" in an ESM bundler results in duplicate code, since the CommonJS module is used by "batchingForReactDom".

webpack-bundle-analyzer in production:

![image](https://user-images.githubusercontent.com/6439050/88342396-f1e37380-cd04-11ea-9a64-0524cd3c2c1b.png)

## How to reproduce the issue

```tsx
import * as React from "react"
import * as ReactDOM from "react-dom"
import "mobx-react-lite/batchingForReactDom"
import {useObserver} from "mobx-react-lite"

const Test = () => {
  return useObserver(() => {
    return <h1>Hello World</h1>
  })
}

ReactDOM.render(<Test />, document.getElementById("app"))
```

## Versions

2.0.7

**Comment by FredyC** [comment]
Thanks for the report. We are considering a different approach to this whole thing. Stay tuned :)

**Comment by davezuko** [comment]
Sounds good, thanks @FredyC .

**Comment by mweststrate** [comment]
@davezuko would you mind verifying if `mobx-react-lite@2.2.0-beta.0` solves the problem?

**Comment by mweststrate** [comment]
Should be fixed in 2.2.0

**Comment by davezuko** [comment]
Looks good to me, thanks @mweststrate.
